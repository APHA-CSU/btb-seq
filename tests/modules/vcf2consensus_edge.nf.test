nextflow_process {

    name "Test Process vcf2Consensus"
    script "main.nf"
    process "VCF2CONSENSUS"

    test("Should produce a masked consensus fasta file from edge-cases input vcf and bed files") {

        when {
            params {
                ref = "${projectDir}/tests/data/edge-cases.fas"
                min_freq_alt = 0.8
                outdir = "${env('PWD')}"
                today = ""
                dataDir = "test"
            }
            process {
                """
                input[0] = params.ref
                input[1] = ["edge-cases",
                            file("${projectDir}/tests/data/edge-cases.mask.bed"),
                            file("${projectDir}/tests/data/edge-cases.nonmasked-regions.bed"),
                            file("${projectDir}/tests/data/edge-cases.vcf.gz"), 
                            file("${projectDir}/tests/data/edge-cases.vcf.gz.csi")]
                input[2] = params.min_freq_alt
                input[3] = params.outdir
                input[4] = params.today
                input[5] = params.dataDir
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.consensus).match()
            assert path(process.out.nCount[0]).readLines().contains('Sample,Ncount,ResultLoc') 
        }

    }

}