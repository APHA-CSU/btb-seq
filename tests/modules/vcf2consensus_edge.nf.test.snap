{
    "Should produce a masked consensus fasta file from edge-cases input vcf and bed files": {
        "content": [
            [
                [
                    "edge-cases",
                    "edge-cases_consensus.fas:md5,ecd4a72e678ab69391155a45807f928e",
                    "edge-cases_snps.tab:md5,e4df5486cba52bbac689745781d6b2ea"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-12T07:59:57.921467934"
    }
}